include/RAJA/pattern/forall.hpp

```diff
@@ -168,13 +168,21 @@ RAJA_INLINE concepts::enable_if<
     type_traits::is_range<Container>>
 forall(ExecutionPolicy&& p, Container&& c, LoopBody&& loop_body)
 {
+  util::PluginContext context{util::make_context<camp::decay<ExecutionPolicy>>()};
+  util::callPreCapturePlugins(context);
 
   using RAJA::internal::trigger_updates_before;
   auto body = trigger_updates_before(loop_body);
 
+  util::callPostCapturePlugins(context);
+
+  util::callPreLaunchPlugins(context);
+
   forall_impl(std::forward<ExecutionPolicy>(p),
               std::forward<Container>(c),
               body);
+
+  util::callPostLaunchPlugins(context);
```

Comment:
I moved the plugins up here to be able to split out the capture and launch cleanly, but I have some questions about what's going on in this file.

Reply:
I've taken a closer look and moved the plugins into a middle layer so they always get called once.